File: jvm-runtime/testdata/java/javaserver/src/main/java/xyz/block/ftl/enums/Verbs.java
package xyz.block.ftl.enums;

import java.util.Map;

import com.fasterxml.jackson.databind.JsonNode;

import ftl.javaserver.Shape;
import ftl.javaserver.StringEnumVerbClient;
import xyz.block.ftl.Export;
import xyz.block.ftl.Verb;

public class Verbs {

    @Export
    @Verb
    public String anyInput(JsonNode node) {
        return node.get("name").asText();
    }

    @Export
    @Verb
    public Object anyOutput(String name) {
        return Map.of("name", name);
    }

    @Export
    @Verb
    public ColorWrapper valueEnumVerb(ColorWrapper color) {
        return color;
    }

    @Export
    @Verb
    public ShapeWrapper stringEnumVerb(ShapeWrapper shape) {
        return shape;
    }

    @Export
    @Verb
    public AnimalWrapper typeEnumVerb(AnimalWrapper animal) {
        return animal;
    }

    @Export
    @Verb
    public ftl.javaserver.ShapeWrapper localVerbCall(StringEnumVerbClient client) {
        return client.stringEnumVerb(new ftl.javaserver.ShapeWrapper(Shape.SQUARE));
    }
}

File: jvm-runtime/testdata/kotlin/kotlinserver/src/main/kotlin/xyz/block/ftl/test/Verbs.kt
package xyz.block.ftl.test

import ftl.kotlinserver.StringEnumVerbClient
import xyz.block.ftl.Export
import xyz.block.ftl.Verb

class Verbs {


  @Export
  @Verb
  fun valueEnumVerb(color: ColorWrapper): ColorWrapper {
    return color
  }

  @Export
  @Verb
  fun stringEnumVerb(shape: ShapeWrapper): ShapeWrapper {
    return shape
  }

  @Export
  @Verb
  fun typeEnumVerb(animal: AnimalWrapper): AnimalWrapper {
    return animal
  }

  @Export
  @Verb
  fun localVerbCall(client: StringEnumVerbClient): ftl.kotlinserver.ShapeWrapper {
     return client.stringEnumVerb(ftl.kotlinserver.ShapeWrapper(ftl.kotlinserver.Shape.Square))
  }

}
